src/agent/configuration_parser/src/configuration_parser.cpp

```diff
@@ -78,6 +55,17 @@ namespace configuration
         }
     }
 
+    void ConfigurationParser::LoadLocalConfig(const std::filesystem::path& configFile)
+    {
+        LogDebug("Loading local config file: {}.", configFile.string());
+        m_config = YAML::LoadFile(configFile.string());
+
+        if (!m_config.IsMap() && !m_config.IsSequence())
```

Comment:
What about function `isValidYamlFile`?

Reply:
Done, this method(`isValidYamlFile`) is now being used to validate it before loading the file.